e2e/tests/transfer/base_test.go

```diff
@@ -417,21 +417,19 @@ func (s *TransferTestSuite) TestMsgTransfer_WithMemo() {
 		transferTxResp, err := s.Transfer(ctx, chainA, chainAWallet, channelA.PortID, channelA.ChannelID, testvalues.DefaultTransferAmount(chainADenom), chainAAddress, chainBAddress, s.GetTimeoutHeight(ctx, chainB), 0, "memo")
 		s.Require().NoError(err)
 
-		if !memoFeatureRelease.IsSupported(chainAVersion) {
-			t.Logf("unsupported")
+		if memoFeatureRelease.IsSupported(chainAVersion) {
+			s.AssertValidTxResponse(transferTxResp)
+		} else {
 			s.Require().Equal(uint32(2), transferTxResp.Code)
-			s.Require().Contains("errUnknownField", transferTxResp.RawLog)
-
-			// transfer not sent, end test
-			return
+			s.Require().Contains(transferTxResp.RawLog, "errUnknownField")
 		}
-
-		t.Logf("supported")
-		// sender chain supports feature
-		s.AssertValidTxResponse(transferTxResp)
-
 	})
 
+	if !memoFeatureRelease.IsSupported(chainAVersion) {
+		// transfer not sent, end test
+		return
```

Comment:
since the checks are run in `t.Run()`, we need a check here to properly return. I left the code as is because I think it is nice doing assertions in t.Run()